storybookjs/storybook
# Storyshots: Add `storyFilterFunction` option

Issue: none

## What I did

Exposing a filter function is more flexible for users and they can
finetune the matching to their liking. Initially I thought it could
deprecate the `storyNameRegex` and `storyKindRegex`, because we could
just create a default filter function which applies those two.

Our use case for this filter function is splitting the story shots based
on an arbitrary count of machines doing the tests in CI. But it will
also make it possible for people just to implement their test matching
to their liking.

## How to test

-   Is this testable with Jest or Chromatic screenshots?
      
    > we could adjust the tests in `storyshots-puppeteer`
-   Does this need a new example in the kitchen sink apps?

    > I don't know
-   Does this need an update to the documentation?

    > Added to the README.

If your answer is yes to any of these, please make sure to include it in your PR.
